snuba/migrations/backfill_errors.py

```diff
@@ -6,462 +6,169 @@
-            ip_v4 = str(ip_address)
-        elif ip_address.version == 6:
-            ip_v6 = str(ip_address)
-
-    # Extract promoted contexts
-    trace_id = None
-    span_id = None
-
-    try:
-        trace_idx = event.contexts_key.index("trace")
-        transaction_ctx = json.loads(event.contexts_value[trace_idx])
-    except ValueError:
-        transaction_ctx = {}
-
-    if transaction_ctx.get("trace_id", None):
-        trace_id = UUID(transaction_ctx["trace_id"])
-    if transaction_ctx.get("span_id", None):
-        span_id = int(transaction_ctx["span_id"], 16)
-
-    error = Error(
-        UUID(event.event_id),
-        event.project_id,
-        event.timestamp,
-        event.platform or "",
-        event.environment,
-        event.release,
-        event.dist,
-        ip_v4,
-        ip_v6,
-        event.user or "",
-        event.user_id,
-        event.user_name,
-        event.user_email,
-        event.sdk_name,
-        event.sdk_version,
-        event.http_method,
-        event.http_referer,
-        event.tags_key,
-        event.tags_value,
-        event.contexts_key,
-        event.contexts_value,
-        event.transaction_name or "",
-        span_id,
-        trace_id,
-        event.partition or 0,
-        event.offset or 0,
-        event.message_timestamp,
-        event.retention_days,
-        event.deleted,
-        event.group_id,
-        UUID(event.primary_hash),
-        event.received or 0,
-        event.message or "",
-        event.title or "",
-        event.culprit or "",
-        event.level,
-        event.location,
-        event.version,
-        event.type or "",
-        event.exception_stacks_type,
-        event.exception_stacks_value,
-        event.exception_stacks_mechanism_type,
-        event.exception_stacks_mechanism_handled,
-        event.exception_frames_abs_path,
-        event.exception_frames_colno,
-        event.exception_frames_filename,
-        event.exception_frames_function,
-        event.exception_frames_lineno,
-        event.exception_frames_in_app,
-        event.exception_frames_package,
-        event.exception_frames_module,
-        event.exception_frames_stack_level,
-        event.sdk_integrations,
-        event.modules_name,
-        event.modules_version,
-    )
-
-    return {
-        name: value
-        for name, value in zip(Error.field_names(), error._asdict().values())
-    }
-
-
-class MigratedEvent(NamedTuple):
-    event_id: str
-    project_id: int
-    timestamp: datetime
-
-    def __repr__(self) -> str:
-        return f"{self.event_id} {self.project_id} {self.timestamp}"
+from snuba.migrations.operations import InsertIntoSelect
+
+
+def format_date(day: date) -> str:
+    return day.strftime("%Y-%m-%d")
+
+
+def get_monday(day: date) -> date:
+    return day - timedelta(days=day.weekday())
+
+
+WINDOW = timedelta(days=7)
+BEGINNING_OF_TIME = get_monday((datetime.utcnow() - timedelta(days=90)).date())
+
+
+COLUMNS = [
```

Comment:
This is missing the geographical information.
For new events managed by the errors consumer we pick the geographical information from the events and put them into contexts. They are not present in contexts otherwise. 
https://github.com/getsentry/snuba/blob/master/snuba/datasets/errors_processor.py#L65-L67
This means that you need to recover them from the `geo_country_code`, `geo_region` and `geo_city` from the events table otherwise they are lost.